src/indexes.ts

```diff
@@ -310,22 +310,26 @@ class Index<T = Record<string, any>> {
    * @memberof Index
    * @method getDocuments
    * @template T
-   * @param {GetDocumentsParams<T>} options? Options to browse the documents
-   * @returns {Promise<GetDocumentsResponse<T>>} Promise containing Document responses
+   * @param {DocumentsParams<T>} options? Options to browse the documents
+   * @returns {Promise<Result<Documents<T>>>} Promise containing Document responses
    */
   async getDocuments<T = Record<string, any>>(
-    options?: GetDocumentsParams<T>
-  ): Promise<GetDocumentsResponse<T>> {
+    options?: DocumentsParams<T>
+  ): Promise<Result<Documents<T>>> {
     const url = `indexes/${this.uid}/documents`
-    let attr
-    if (options !== undefined && Array.isArray(options.attributesToRetrieve)) {
-      attr = options.attributesToRetrieve.join(',')
+
+    let fields = undefined
```

Comment:
This is something contradictory don't you think? To start a variable with undefined :)

Reply:
This would be the alternative: 
```js
const fields = (() => {
      if (Array.isArray(options?.fields)) {
        return options?.fields?.join(',')
      }
      return undefined
    })()
 ```
 
 I need to define the fields on the highest scope so that the changes made inside the if are done on a variable that is accessible on that scope.